src/lightning_lite/plugins/collectives/collective.py

```diff
@@ -0,0 +1,157 @@
+import datetime
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+import torch
+from torch.distributed import ProcessGroup, ReduceOp
+
+
+class Collective(ABC):
+    def __init__(self, delayed_init: bool, *init_args, **init_kwargs):
+        if delayed_init:
+            self._group = None
+            self._init_args = init_args
+            self._init_kwargs = init_kwargs
+        else:
+            self._group = self._init_group_impl(*init_args, **init_kwargs)
```

Comment:
we should define these unconditionally.

And we can rely on the getter for instantaition

```suggestion
        self._group = None
        self._init_args = init_args
        self._init_kwargs = init_kwargs
        if intantiate_pg:
            _ = self.group 
```